docs/DEVELOPING.md

```diff
@@ -277,6 +272,177 @@ See [Troubleshooting](https://www.elastic.co/guide/en/enterprise-search/master/c
+    ```
+    User.Read
+    ```
+  - Finally, **Grant admin consent**.
+
+    Use the Grant Admin Consent link from the permissions screen.
+  - At last also save the tenant name (i.e. Domain name) of Azure platform
+
+
+### SharePoint Online permissions required to run the connector:
+
+   Refer to the following documentation for setting [SharePoint Permissions](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs)
+
+   - To set `DisableCustomAppAuthentication` to false we need to connect to SharePoint using Windows PowerShell and run set-spotenant -DisableCustomAppAuthentication $false
+   - To assign full permissions to the tenant in SharePoint Online, in your browser, go to the tenant URL. For example, go to `https://<office_365_admin_tenant_URL>/_layouts/15/appinv.aspx` such as `https://contoso-admin.sharepoint.com/_layouts/15/appinv.aspx`. The SharePoint admin center page appears.
+       - In the App ID box, enter the application ID that you recorded earlier, and then click Lookup. In the Title box, the name of the application appears.
+       - In the App Domain box, type tenantname.onmicrosoft.com
+       - In the App's Permission Request XML box, type the following XML string:
+           ```
+            <AppPermissionRequests AllowAppOnlyPolicy="true">
+              <AppPermissionRequest Scope="http://sharepoint/content/tenant" Right="FullControl" />
+              <AppPermissionRequest Scope="http://sharepoint/social/tenant" Right="Read" />
+            </AppPermissionRequests>
+            ```
+
+### Compatibility
+
+  Both SharePoint Online and SharePoint server are supported by the connector.
+
+  For SharePoint server, below mentioned versions are compatible with Elastic connector frameworks:
+  - SharePoint 2013
+  - SharePoint 2016
+  - SharePoint 2019
+
+### Configuration
+
+The following configuration fields need to be provided for setting up the connector:
+
+##### `data_source`
+
+Determines the SharePoint platform type. `SHAREPOINT_ONLINE` if SharePoint cloud and `SHAREPOINT_SERVER` if SharePoint Server.
+
+##### `username`
+
+The username of the account for SharePoint Server.
+
+**Note:** `username` is not needed for SharePoint Online.
+
+##### `password`
+
+The password of the account to be used for the SharePoint Server.
+
+**Note:** `password` is not needed for SharePoint Online.
+
+##### `client_id`
+
+The client id to authenticate with SharePoint Online.
+
+##### `client_secret`
+
+The secret value to authenticate with SharePoint Online.
+
+##### `tenant`
+
+The tenant name to authenticate with SharePoint Online.
+
+##### `tenant_id`
+
+The tenant id to authenticate with SharePoint Online.
+
+##### `host_url`
+
+The server host url where the SharePoint is hosted. Examples:
+
+  - `https://192.158.1.38:8080`
+  - `https://test_user.sharepoint.com`
+
+##### `site_collections`
```

Comment:
Is there a syntax for "all"? I tried `""` and `*` and neither worked. If this is possible, we should document it here.

Reply:
No, currently we don't have any option for all.